src/common/ittnotify.cpp

```diff
@@ -43,16 +43,34 @@ bool get_itt(__itt_task_level level) {
 namespace {
 
 thread_local primitive_kind_t thread_primitive_kind;
+thread_local const char *thread_primitive_info;
+thread_local const char *thread_primitive_op;
+__itt_string_handle *thread_primitive_meta_fmt
+        = __itt_string_handle_create("%s");
 
-__itt_domain *itt_domain() {
-    static __itt_domain *d = __itt_domain_create("dnnl::primitive::execute");
+__itt_domain *itt_domain(const char *prim_op) {
+    if (std::strcmp(prim_op, "exec") == 0) {
+        static __itt_domain *d_exec
+                = __itt_domain_create("dnnl::primitive::execute");
+        return d_exec;
+    }
+
+    if (std::strcmp(prim_op, "create") == 0) {
+        static __itt_domain *d_create
+                = __itt_domain_create("dnnl::primitive::create");
+        return d_create;
+    }
+
+    static __itt_domain *d = __itt_domain_create(prim_op);
     return d;
 }
 
 } // namespace
 
-void primitive_task_start(primitive_kind_t kind) {
+void primitive_task_start(
+        primitive_kind_t kind, const char *pd_info, const char *primitive_op) {
```

Comment:
`primitive_op` name is a bit misleading, as this suggest it is related to primitive_kind somehow. Maybe use `itt_grouping` or `log_kind` to align with verbose naming?